### DISCUSSION TITLE
Meeting note #9, 2023-01-03 final

### DISCUSSION THREAD

**Discussion by jongfeel** [wisdom08/dev-community#46] [General]
## Communication에 대해

제가 제시한 DB 테이블 설계 부분에 대해서 다시 리뷰를 진행해 보시라고 했었는데,
실제로 확인해 보고 문제가 없었다면 그 내용을 공유하고 확인할 수 있게 만들어 주는 게
같이 내용을 공유하고 얘기하는 사람과 해야 하는 중요한 부분입니다.
우리는 그걸 커뮤니케이션 혹은 협업을 잘하는 거라고 얘기합니다.

개발을 어떻게 하면 되겠다 생각은 잘 하고 계시고
그 생각이 코드로 옮겨지는 데 까지 무리는 없는 것 같은데
그걸 문서로 표현하고 다른 사람과 공유하고 소통하는데 있어서는 많이 부족한 건 사실입니다.

마지막 멘토링이었지만 향후에도 아래 두 가지를 잘 훈련해 보셨으면 좋겠습니다.

- 코드를 작성하기 전에 구현 작전에 대한 문서를 잘 적어보자
- 문서와 코드의 내용 일치를 위해 문서와 코드 두 부분을 지속적으로 업데이트 해 가면서 진행해야 한다.

## 설계를 해보기 위한 방법

여기서 구현 작전에 대한 건 아래와 같습니다. 데이터 중심 처리 방식이죠.

- 데이터: 데이터가 어떻게 생성되고, 어디에 저장되서, 어떻게 전달하거나 받고, 어떻게 표현할 것인가를 기록
- 프로세스: 데이터를 가지고 어떤 처리를 하는가, 주로 원시 데이터를 통해 원하는 처리를 위한 함수 구현으로 이해. 구체적인 함수 구현을 하는 게 아니라 입력, 출력, 처리에 대한 정의만 진행
- 결과: 데이터를 가지고 프로세스를 거치면 어떤 기대 결과가 나오는지에 대한 기록
- 검증: 데이터-프로세스-결과에 대한 검증은 이런 테스트 과정으로 확인할 수 있고 검증 가능하다는 부분에 대한 기록

입니다.

어떤 피처(feature)가 기능(function)을 구현하느냐에 따라 분량이 달라질 수는 있지만
꼭 템플릿에 맞춰서 적어 넣듯이 하는 것만 아니면 됩니다.

지금은 잘 안되고 못한다고 하더라도 자책하지 마시고
이걸 할 수 있어야 소프트웨어 엔지니어라고 부를 수 있다 정도만 생각하시면 됩니다.

## 끝으로

자신의 생각의 변화를 잘 기록하는 게 많이 중요합니다.
향후 동료와 커뮤니케이션을 위해서도 필요하고
자기 자신의 성장과 발전을 위한 회고를 위해서도 필요합니다.

정말 꼭 훈련을 조금씩 하셔서, 커뮤니케이션이라는 걸 할 수 있는 개발자가 되셨으면 좋겠습니다.

**Comment by wisdom08** [comment]
네 멘토님 작년부터 정말 감사했습니다. 
끝이 날까? 하는 생각도 들었지만 끝은 났네요 ㅎㅎ 
멘토님과 대화하는 시간이 뜨끔하는 시간도 많았지만.. 돌이켜보면 정말 즐거웠습니다!

멘토링 노트를 처음부터 다시 다 봤는데 저에게 전달하려고 해주신 건 결국에는
커뮤니케이션 하는 개발자가 되어야 한다!! 라는 것 같습니다. 
커뮤니케이션 부분에 대해서는 욕심부리지는 않고 조금씩 의식적으로 훈련 해보겠습니다.

감사합니다🙇🏻‍♀️
